projects/storefrontlib/src/cms-components/checkout/services/express-checkout.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from '@angular/core';
+import { combineLatest, of } from 'rxjs';
+import { filter, map, switchMap, withLatestFrom } from 'rxjs/operators';
+
+import {
+  Address,
+  CheckoutDeliveryService,
+  UserAddressService,
+  UserPaymentService,
+  PaymentDetails,
+  DeliveryMode,
+  CheckoutPaymentService,
+} from '@spartacus/core';
+import { CheckoutConfigService } from './checkout-config.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExpressCheckoutService {
+  shippingAddress$;
+  deliveryMode$;
+  paymentMethod$;
+
+  constructor(
+    protected userAddressService: UserAddressService,
+    protected checkoutDeliveryService: CheckoutDeliveryService,
+    protected checkoutPaymentService: CheckoutPaymentService,
+    protected userPaymentService: UserPaymentService,
+    protected checkoutConfigService: CheckoutConfigService
+  ) {
+    this.setShippingAddress();
+    this.setDeliveryMode();
+    this.setPaymentMethod();
+  }
+
+  protected setShippingAddress() {
+    this.userAddressService.loadAddresses();
+    this.shippingAddress$ = this.userAddressService
+      .getAddressesLoadedSuccess()
+      .pipe(
+        filter(Boolean),
+        switchMap(() => this.userAddressService.getAddresses()),
+        map((addresses: Address[]) => {
+          const defaultAddress =
+            addresses.find(address => address.defaultAddress) || addresses[0];
+          if (defaultAddress) {
+            this.checkoutDeliveryService.setDeliveryAddress(defaultAddress);
```

Comment:
I would only return true after deliveryAddress is set. There might be the case when you would check supported modes before deliveryAddress would set and then with an empty list of delivery modes express checkout would fail

Reply:
there is filter
```typescript
filter(([deliveryModes]: [DeliveryMode[], string]) =>
              Boolean(deliveryModes.length)
            ),
```